hsm-gateway-applications/node/src/hsm-sample.ts

```diff
@@ -0,0 +1,151 @@
+    const hsmSignerFactory = signers.newHSMSignerFactory(findSoftHSMPKCS11Lib());
+    const credentials = await fs.promises.readFile(certPath);
+
+    // Get the signer function and a close function. The close function closes the signer
+    // once there is no further need for it.
+    const {signer, close} = await newHSMSigner(hsmSignerFactory, credentials.toString());
+
+    const gateway = connect({
+        client,
+        identity: {mspId, credentials},
+        signer,
+    });
+
+    try {
+        await exampleTransaction(gateway);
+        console.log();
+        console.log('Node HSM sample completed successfully');
+    } finally {
+        gateway.close();
+        client.close();
+
+        // close the HSM Signer
+        close();
+    }
+}
+
+async function exampleTransaction(gateway: Gateway):Promise<void> {
+    const network = gateway.getNetwork('mychannel');
+    const contract = network.getContract('basic');
+
+    console.log('\n--> Submit Transaction: CreateAsset, creates new asset with ID, Color, Size, Owner and AppraisedValue arguments');
+
+    await contract.submitTransaction(
+        'CreateAsset',
+        assetId,
+        'yellow',
+        '5',
+        'Tom',
+        '1300',
+    );
+
+    console.log('*** Transaction committed successfully');
+
+    console.log('\n--> Evaluate Transaction: ReadAsset, function returns asset attributes');
+
+    const resultBytes = await contract.evaluateTransaction('ReadAsset', assetId);
+
+    const resultJson = utf8Decoder.decode(resultBytes);
+    const result = JSON.parse(resultJson);
+    console.log('*** Result:', result);
+}
+
+async function newGrpcConnection(): Promise<grpc.Client> {
+    const tlsRootCert = await fs.promises.readFile(tlsCertPath);
+    const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
+
+    return new grpc.Client(peerEndpoint, tlsCredentials, {
+        'grpc.ssl_target_name_override': 'peer0.org1.example.com'
+    });
+}
+
+// Create a new HSM Signer
+async function newHSMSigner(hsmSignerFactory: HSMSignerFactory, certificatePEM: string): Promise<HSMSigner> {
+    const ski = getSKIFromCertificate(certificatePEM);
+
+    // Options for the signer based on using SoftHSM with Token initialized as follows
+    // softhsm2-util --init-token --slot 0 --label "ForFabric" --pin 98765432 --so-pin 1234
+    const hsmSignerOptions: HSMSignerOptions = {
+        label: 'ForFabric',
+        pin: '98765432',
+        identifier: ski
+    }
+    return hsmSignerFactory.newSigner(hsmSignerOptions);
+}
+
+// Utility to find the SoftHSM PKCS11 library as it's location can vary based on
+// operating system and version
+function findSoftHSMPKCS11Lib(): string {
+    const commonSoftHSMPathNames = [
+        '/usr/lib/softhsm/libsofthsm2.so',
+        '/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so',
+        '/usr/local/lib/softhsm/libsofthsm2.so',
+        '/usr/lib/libacsp-pkcs11.so',
+    ];
+
```

Comment:
Need to consider PKCS11_LIB environment variable is set to specify a non-standard library location.
```typescript
    const pkcs11lib = process.env['PKCS11_LIB'];
    if (pkcs11lib) {
        commonSoftHSMPathNames.push(pkcs11lib);
    }

```